metaflow/plugins/test_unbounded_foreach_decorator.py

```diff
@@ -60,13 +66,43 @@ def step_init(
     ):
         self.environment = environment
 
+    def task_pre_step(
+        self,
+        step_name,
+        task_datastore,
+        metadata,
+        run_id,
+        task_id,
+        flow,
+        graph,
+        retry_count,
+        max_user_code_retries,
+        ubf_context,
+        inputs,
+    ):
+        if ubf_context == UBF_CONTROL:
+            metadata.register_metadata(
+                run_id,
+                step_name,
+                task_id,
+                [
+                    MetaDatum(
+                        field="internal_task_type",
+                        value=CONTROL_TASK_TAG,
+                        type="internal_task_type",
+                        tags=["attempt_id:{0}".format(0)],
+                    )
+                ],
+            )
+        self.input_paths = [obj.pathspec for obj in inputs]
```

Comment:
This line can probably move into the ubf_context if block as it is then only used for the control function.

Reply:
done